### ISSUE TITLE
[Question] Firebase Auth - How to handle untrusted providers when signing in ?

### ISSUE THREAD

**Issue by Kapusch** [TobiasBuchholz/Plugin.Firebase#56]
Hi @TobiasBuchholz , I noticed a problem which can be annoying for my users : 

> Considering a user "test123@gmail.com" who registered to my app using Google Sign In. If one day this same user decides to use Facebook Sign In with the same email address, he will get stuck on the Web flow for Facebook authentication with a meaningless message saying : "Inactive App : This app is not usable for the moment. You will be able to authenticate once the app is reactivated."
<img width="379" alt="image" src="https://user-images.githubusercontent.com/9576733/182044150-5a4d249b-221b-4009-861f-3a1e2ba87eaf.png">

Indeed, in my Firebase project I have allowed [multiple accounts with the same email address](https://support.google.com/firebase/answer/9134820) and this means a user will be always linked to one unique Firebase user account even if using different providers with the same email address.

However, Firebase separates _trusted_ from _untrusted_ providers. And there comes the actual problem : [Facebook is not considered by Firebase as a trusted provider](https://firebase.google.com/docs/auth/users#verified_email_addresses).

To summarize, user will not get any issue if using Facebook Sign In, until he authenticates with a truster provider like Google or Apple, and here I'm afraid I can lose some users.

## Conclusion

When user goes for Facebook Sign In, I would like your library to be able to fetch the list of enabled providers associated to his email address so that it could raise a "WeakProvider" exception. Once handled in my application code, I could provide support and redirect him to another way of signing in.

**Comment by Kapusch** [comment]
Note : I have found different threads on the Web mentioning this issue, and [here](https://stackoverflow.com/a/44528975/4585688) there's actually a diagram which can help understanding a possible solution to implement : 
<img width="971" alt="image" src="https://user-images.githubusercontent.com/9576733/182044488-99f2f977-7a22-43ae-aa58-14851cdaf670.png">

**Comment by Kapusch** [comment]
Hi @TobiasBuchholz , I just would like to inform you that I'm planning to work on this one.
I will keep you informed further this week

**Comment by Kapusch** [comment]
Finally, I was not able to retrieve the email used by the user in the Facebook sign in flow, so I simply "catched" the right error in order to raise a specific `CrossFirebaseAuthException` accordingly.

An improvement could be to retrieve user email in the catched exception, and then fetch the associated sign in methods to help the user to understand "which trusted provider he must use", but that's much complex and I don't have the time for this.

So I will raise a PR for my solution, feel free to improve it !

**Comment by Kapusch** [comment]
#64

**Comment by TobiasBuchholz** [comment]
Thanks a lot for your efforts @Kapusch! Your PR is merged now and available in the plugin version 1.1.4 :)
